FROM ubuntu:18.04

RUN apt-get update \
    && apt-get -y install software-properties-common \
    autoconf \
    automake \
    build-essential \
    cmake \
    pkg-config \
    wget \
    python-pip \
    libhdf5-dev \
    libhdf5-serial-dev \
    hdf5-tools\
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*

RUN wget https://github.com/lhelontra/tensorflow-on-arm/releases/download/v1.11.0/tensorflow-1.11.0-cp27-none-linux_aarch64.whl \
    && pip install tensorflow-1.11.0-cp27-none-linux_aarch64.whl \
    && rm tensorflow-1.11.0-cp27-none-linux_aarch64.whl \
    && rm -rf .cache

RUN pip install rfc3339 grpcio googleapis-common-protos jupyter
ADD . /usr/src/app/github.com/kubeflow/katib
WORKDIR /usr/src/app/github.com/kubeflow/katib/cmd/metricscollector/v1beta1/tfevent-metricscollector/
RUN pip install --no-cache-dir -r requirements.txt
ENV PYTHONPATH /usr/src/app/github.com/kubeflow/katib:/usr/src/app/github.com/kubeflow/katib/pkg/apis/manager/v1beta1/python:/usr/src/app/github.com/kubeflow/katib/pkg/metricscollector/v1beta1/tfevent-metricscollector/:/usr/src/app/github.com/kubeflow/katib/pkg/metricscollector/v1beta1/common/
ENTRYPOINT ["python", "main.py"]
